BasiliskII/src/BeOS/extfs_beos.cpp

```diff
@@ -458,3 +459,31 @@ bool extfs_rename(const char *old_path, const char *new_path)
 {
 	return rename(old_path, new_path) == 0;
 }
+
+/*
+ *  Strings (filenames) conversion
+ */
+
+// Convert from the host OS filename encoding to MacRoman
+const char *host_encoding_to_macroman(const char *filename)
+{
+	int32 state = 0;
+	static char buffer[512];
+	int32 size = 512;
+	int32 insize = strlen(filename);
+	convert_from_utf8(B_MAC_ROMAN_CONVERSION, filename, &insize, buffer, &size, &state);
+	buffer[size] = 0;
+	return buffer;
+}
+
+// Convert from MacRoman to host OS filename encoding
+const char *macroman_to_host_encoding(const char *filename)
+{
+	int32 state = 0;
+	static char buffer[512];
+	int32 insize = strlen(filename);
+	int32 size = 512;
```

Comment:
I think this needs to be 511, else if 512 bytes will be written, line 487 will write pas the buffer.

Please change to:

int32 size = sizeof(buffer) -1;

And also do the same above.